File: Tree/617. Merge Two Binary Trees/mergeTrees.go
package main

// TreeNode 定义二叉树节点结构
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

// mergeTrees 合并两棵二叉树
func mergeTrees(root1 *TreeNode, root2 *TreeNode) *TreeNode {
	if root1 != nil && root2 != nil {
		result := &TreeNode{Val: root1.Val + root2.Val}
		result.Left = mergeTrees(root1.Left, root2.Left)
		result.Right = mergeTrees(root1.Right, root2.Right)
		return result
	} else if root1 != nil && root2 == nil {
		return root1 // 直接返回root1，因为root2为空，无需合并
	} else if root1 == nil && root2 != nil {
		return root2 // 直接返回root2，因为root1为空，无需合并
	} else {
		// 当root1和root2都为null时，可以返回一个null或者一个新的空节点
		return nil // 或者 return &TreeNode{Val: 0}
	}
}

File: Tree/617. Merge Two Binary Trees/merge_trees.rs
#[derive(Debug, PartialEq, Eq)]
pub struct TreeNode {
    pub val: i32,
    pub left: Option<Rc<RefCell<TreeNode>>>,
    pub right: Option<Rc<RefCell<TreeNode>>>,
}

impl TreeNode {
    #[inline]
    pub fn new(val: i32) -> Self {
        TreeNode {
            val,
            left: None,
            right: None,
        }
    }
}

use std::rc::Rc;
use std::cell::RefCell;

impl Solution {
    pub fn merge_trees(root1: Option<Rc<RefCell<TreeNode>>>, root2: Option<Rc<RefCell<TreeNode>>>) -> Option<Rc<RefCell<TreeNode>>> {
        match (root1, root2) {
            (Some(node1), Some(node2)) => {
                let mut node1 = node1.borrow_mut();
                let mut node2 = node2.borrow_mut();
                let merged_node = TreeNode {
                    val: node1.val + node2.val,
                    left: Self::merge_trees(node1.left.take(), node2.left.take()),
                    right: Self::merge_trees(node1.right.take(), node2.right.take()),
                };
                Some(Rc::new(RefCell::new(merged_node)))
            }
            (Some(node1), None) => Some(node1),
            (None, Some(node2)) => Some(node2),
            (None, None) => None,
        }
    }
}